app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java

```diff
@@ -343,10 +456,63 @@ public static Graph addNodesToGraph(String[] contents)
                 if (i == 0)
                     current = coordinatesToInsert.getHead();
                 if (current != null)
-                    returnMe.insertVertex((LatLng)current.getEle());
+                    if (i < nmbClassNodes) // i < nmbClassNodes the condition causing the error
+                    {
+                        returnMe.insertVertex((LatLng)current.getEle(), 0); //insert a class node
+                    } else {
+                        returnMe.insertVertex((LatLng)current.getEle(), 1); //insert a hall node
+                    }
                 current = current.getNext();
             }
         }
         return returnMe;
     }
+
+    public static LatLng readClassCoordinate(String currentLine)
```

Comment:
Where does String currentLine come from? When is this method called and by which class?

Reply:
This is a helper method that is used inside the addNodesToGraph method above. 
currentLine is passed in when reading from a file of Nodes

Reply:
Shouldn't helper methods be private?

Reply:
yes i will change them to private!